matheus_massula_test/test/unit/qr_code_web_client_test.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:matheus_massula_test/models/remote/qr_code_seed.dart';
+import 'package:mockito/mockito.dart';
+import '../mocks/qr_code_web_client_mock.dart';
+
+void main() {
+  test('Should retrieve successfully QR Code Seed', () async {
+    final QRCodeWebClientMock qrCodeWebClientMock = QRCodeWebClientMock();
+
+    when(qrCodeWebClientMock.getQRCode()).thenAnswer(
+      (_) async => QRCodeSeed(seed: 'random data', expiresAt: DateTime.now())
+    );
+
+    final result = qrCodeWebClientMock.getQRCode();
+
+    expect(result, isA<Future<QRCodeSeed>>());
+  });
+
+  test('Should not retrieve QR Code Seed', () async {
+    final QRCodeWebClientMock qrCodeWebClientMock = QRCodeWebClientMock();
+
+    when(qrCodeWebClientMock.getQRCode()).thenAnswer(
+      (_) async => throw Exception()
+    );
+
+    final result = qrCodeWebClientMock.getQRCode();
+
+    expect(result, throwsA(isA<Exception>()));
+  });
+}
```

Comment:
issue: If these are intended to test `QRCodeWebClient` then that class should not be mocked here. Fake JSON data should be returned from a mock HTTP client that is injected into a real instance of `QRCodeWebClient` instead. Currently these tests aren't verifying any of your actual code.

Reply:
I mocked the client but I had an issue related to the null safety feature.
See the code:

...
```
class ClientMock extends Mock implements Client { }

test('Should retrieve successfully QR Code Seed', () async {
  final ClientMock clientMock = ClientMock();
  final QRCodeWebClient qrCodeWebClient = QRCodeWebClient(clientMock);
  
  when(clientMock.get(Uri.parse(url + 'default/random-qr-seed_seed'))).thenAnswer(
    (_) async => Response('{"seed": "random data", "expiresAt": ${DateTime.now()}}', 200)
  );

  final result = qrCodeWebClient.getQRCode();

  expect(result, isA<Future<QRCodeSeed>>());
});
```

[here is](https://stackoverflow.com/questions/67371802/dart-type-null-is-not-a-subtype-of-type-futurestring-in-mockito) the problem I'm passing through.

Reply:
`mockito` underwent breaking changes in order to migrate to null safety. [This writeup](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md) will help you resolve the issue you're seeing.